crates/swc_ecma_transforms_react/src/lib.rs

```diff
@@ -39,35 +41,35 @@ mod refresh;
 pub fn react<C>(
     cm: Lrc<SourceMap>,
     comments: Option<C>,
-    mut options: Options,
+    options: Options,
     top_level_mark: Mark,
     unresolved_mark: Mark,
 ) -> impl Pass
 where
     C: Comments + Clone,
 {
-    let Options { development, .. } = options;
-    let development = development.unwrap_or(false);
+    let development = options.common.development.into_bool();
+
+    let (auto_config, classic_config) = match parse_directives(options.runtime, comments.clone()) {
```

Comment:
Is it safe to access comments outside the visitor? 
I've observed in the test suite that parser execution can be delayed, resulting in empty comments being read.

Reply:
Yeap, it's safe to access. It's not scoped tls so it does not matter